cmd/kubelet/app/server.go

```diff
@@ -598,6 +623,83 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 	return nil
 }
 
+func updateTransport(clientConfig *restclient.Config, clientCertificateManager certificate.Manager) error {
+	pool := x509.NewCertPool()
+	if len(clientConfig.CAFile) > 0 {
+		if certs, err := certutil.NewPool(clientConfig.CAFile); err != nil {
+			return fmt.Errorf("unable to load the CA from %q: %v", clientConfig.CAFile, err)
+		} else {
+			pool = certs
+		}
+	}
+	if len(clientConfig.CAData) > 0 {
+		if certs, err := certutil.ParseCertsPEM(clientConfig.CAData); err != nil {
+			return fmt.Errorf("unable to parse the CA from the data in the config file: %v", err)
+		} else {
+			for _, cert := range certs {
+				pool.AddCert(cert)
+			}
+		}
+	}
+	clientConfig.Transport = utilnet.SetTransportDefaults(&http.Transport{
+		TLSClientConfig: &tls.Config{
+			RootCAs: pool,
+			GetClientCertificate: func(requestInfo *tls.CertificateRequestInfo) (*tls.Certificate, error) {
+				cert := clientCertificateManager.Current()
+				if cert == nil {
+					return nil, fmt.Errorf("no certificate available")
+				}
+				return cert, nil
+			},
+		},
+	})
+	clientConfig.CertData = nil
```

Comment:
CertData,CertFile,KeyData,KeyFile are of a piece... if this is replacing them, I'd expect them to all be cleared together